dev/benchmarks/complex_layout/test/scroll_smooth_measure.dart

```diff
@@ -0,0 +1,231 @@
+
+  Map<String, dynamic> result;
+
+  @override
+  String describeValue(bool value) => value ? 'resampling on' : 'resampling off';
+
+  @override
+  Future<bool> setUp(bool value) async {
+    final bool original = binding.resamplingEnabled;
+    binding.resamplingEnabled = value;
+    return original;
+  }
+
+  @override
+  Future<void> tearDown(bool value, bool memento) async {
+    binding.resamplingEnabled = memento;
+    binding.reportData[value ? 'with resampler' : 'without resampler'] = result;
+  }
+}
+
+Future<void> main() async {
+  final PointerDataTestBinding binding = PointerDataTestBinding();
+  assert(WidgetsBinding.instance == binding);
+  binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.benchmarkLive;
+  binding.reportData ??= <String, dynamic>{};
+  final ResampleFlagVariant variant = ResampleFlagVariant(binding);
+  testWidgets('Smoothness test', (WidgetTester tester) async {
+    app.main();
+    await tester.pumpAndSettle();
+    final Finder scrollerFinder = find.byKey(const ValueKey<String>('complex-scroll'));
+    final ListView scroller = tester.widget<ListView>(scrollerFinder);
+    final ScrollController controller = scroller.controller;
+    final List<int> frameTimestamp = <int>[];
+    final List<double> scrollOffset = <double>[];
+    binding.addPersistentFrameCallback((Duration timeStamp) {
+      if (controller.hasClients) {
+        // This if is necessary because by the end of the test the widget tree
+        // is destroyed.
+        frameTimestamp.add(timeStamp.inMicroseconds);
+        scrollOffset.add(controller.offset);
+      }
+    });
+
+    Duration now() => binding.currentSystemFrameTimeStamp;
+    Future<List<Duration>> scroll() async {
+      // Extra 50ms to avoid timeouts.
+      final Duration startTime = const Duration(milliseconds: 500) + now();
+      final List<Duration> delays = <Duration>[];
+      for (final PointerDataRecord record in dragInputDatas(startTime, tester.getCenter(scrollerFinder))) {
+        await tester.binding.delayed(record.timeStamp - now());
+        // This now measures how accurate the above delayed is.
+        delays.add(now() - record.timeStamp);
+        ui.window.onPointerDataPacket(record.data);
+      }
+      return delays;
+    }
+
+    List<Duration> delays = <Duration>[];
+    for (int n = 0; n < 5; n++) {
+      delays += await scroll();
+    }
+    variant.result = scrollSummary(scrollOffset, delays, frameTimestamp);
+    await tester.pumpAndSettle();
+    scrollOffset.clear();
+    delays.clear();
+    await tester.idle();
+  }, semanticsEnabled: false, variant: variant);
+}
+
+/// Calculates the smoothness measure from `scrollOffset` and `delays` list.
+///
+/// For smoothness it filters the frames with delays too longer to separate the
+/// effect of building a frame too slow. Frames that are built slow are counted
+/// to `dropped_frame_count`.
+///
+/// Smoothness (jerk) is measured by the average of the absolute value for
```

Comment:
"Smoothness (`abs_jerk`) is measured by  the absolute value of ...",
"Its average (`average_abs_jerk`) is calculated for the overall smoothness",
"frames with `abs_jerk` larger than 0.5 is counted to `janky_count`".

Let's also refactor the code to rename `jerk` to `absJerk` to make it consistent with the documentation and report keys.

Reply:
Done.